AndroidSDK/src/com/leanplum/internal/LeanplumEventDataManager.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2017, Leanplum, Inc. All rights reserved.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.leanplum.internal;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+
+import android.database.DatabaseUtils;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteDatabaseCorruptException;
+
+import org.json.JSONObject;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * LeanplumEventDataManager class to work with SQLite.
+ *
+ * @author Anna Orlova
+ */
+public class LeanplumEventDataManager {
+  private static final String DATABASE_NAME = "__leanplum.db";
+  private static final String EVENT_TABLE_NAME = "event";
+  private static final String COLUMN_DATA = "data";
+  private static final String KEY_ROWID = "rowid";
+
+  private static SQLiteDatabase database;
+  private static ContentValues contentValues = new ContentValues();
+
+  public static void init(Context context) {
+    if (database != null) {
+      Log.e("Database is already initialized.");
+      return;
+    }
+    File dbFile = new File(context.getFilesDir(), DATABASE_NAME);
+    boolean needMigration = false;
+    if (!dbFile.exists()) {
+      needMigration = true;
+    }
+    // Create database if needed.
+    try {
+      database = SQLiteDatabase.openDatabase(dbFile.getAbsolutePath(), null,
```

Comment:
Subclass SQLiteOpenHelper to gain support for versioning, thread safety access and initialization. 
https://github.com/android/platform_frameworks_base/blob/master/core/java/android/database/sqlite/SQLiteOpenHelper.java#L191